### ISSUE TITLE
Question about "Digging Into Self-Supervised Monocular Depth Estimation"

### ISSUE THREAD

**Issue by ohwi** [mrharicot/monodepth#176]
I was wondering how is your network trained with stereo and monocular data combined.

Is it the way that is like fine-tune or merging losses?

Thanks

**Comment by mrharicot** [comment]
Hi,
It happens in the loss, instead of only having the forward and backward frames, we also reproject the middle stereo frame with a fixed transformation (horizontal).
As before we then take the per-pixel minimum over those 3 reprojected frames.

**Comment by ohwi** [comment]
Thank you for the answer!
Can I ask one more question?

For the stereo data, the disparity which is the output of the network is used for horizontal shift.

And, for the monocular, the depth calculated as [here](https://github.com/mrharicot/monodepth/issues/167) with the disparity same as the one that used for stereo data is used for reprojection.

Is my understanding right?

**Comment by mrharicot** [comment]
We use the exact same depth-based reprojection code for the monocular and stereo images.
Because of the purely horizontal transformation for the stereo image, it results in a simple horizontal shift (disparity) but for simplicity we use the same code.

**Comment by ricshaw** [comment]
Does this mean that for monocular images, to project forwards and backwards you use a 2D version of your bilinear_samper.py but for a 2D disparity?

**Comment by mrharicot** [comment]
@ricshaw yes!

**Comment by daniyar-niantic** [comment]
@ohwi The implementation is now public at https://github.com/nianticlabs/monodepth2
